### PR TITLE
add support for python 3.9 preview and update dotnetcore 2.1 and 3.1

### PR DESCRIPTION
**Pull request by arroyc**
update python 3.8 to 3.8.3
add support for python 3.9-rc
update dotnetcore 2.1 
update dotnetcore 3.1

### CODE REVIEW

File: images/runtime/dotnetcore/3.1/base.Dockerfile [hunk 1]
```diff
@@ -35,10 +35,10 @@ COPY --from=tools-install /dotnetcore-tools /opt/dotnetcore-tools
 ENV PATH="/opt/dotnetcore-tools:${PATH}"
 
 # Install .NET Core
-ENV DOTNET_VERSION 3.1.3
+ENV DOTNET_VERSION 3.1.4
```
Comment (kichalla) [inline images/runtime/dotnetcore/3.1/base.Dockerfile:38]: Need to update the aspnet core version on line 49 too

File: platforms/dotnet/versionsToBuild.txt [hunk 2]
```diff
@@ -24,6 +24,7 @@
 3.1.102, 9cacdc9700468a915e6fa51a3e5539b3519dd35b13e7f9d6c4dd0077e298baac0e50ad1880181df6781ef1dc64a232e9f78ad8e4494022987d12812c4ca15f29,
 3.1.200, 5b9398c7bfe7f67cd9f38fdd4e6e429e1b6aaac0fe04672be0f8dca26580fb46906fd1d2deea6a7d3fb07d77e898f067d3ac1805fe077dc7c1adf9515c9bc9a9, 
 3.1.201, 934bf29734776331691a4724f2174c4e9d2d1dde160f18397fd01abf0f96f2ec1bdd2874db9f0e25dce6993d527ea9c19031a0e67383fd813dcfcb9552ea0c70,
+3.1.202, b570a423522de777d05de67410cea7d6d54461b7e2b82a18f4ca21093897836447f215b2737df909ee581508fbd1a61993f3ea1d9856b9a883caed3f7086f0a6
```
Comment (kichalla) [inline platforms/dotnet/versionsToBuild.txt:27]: We need a base image build for runtime image also

File: build/__dotNetCoreRunTimeVersions.sh [hunk 3]
```diff
@@ -7,5 +7,5 @@ NET_CORE_APP_20='2.0.9'
 NET_CORE_APP_21='2.1.17'
```
Comment (kichalla) [inline build/__dotNetCoreRunTimeVersions.sh:7]: Seems like even 2.1 version has updates. Since it is LTS version, could you update that too?
https://github.com/dotnet/core/blob/master/release-notes/2.1/releases.json#L133

File: platforms/dotnet/versionsToBuild.txt (hunk 2 above)
Comment (arroyc) [reply platforms/dotnet/versionsToBuild.txt:27]: yes .. i will do that .. currently local build is failing on sha512 verification ..

File: build/__dotNetCoreRunTimeVersions.sh (hunk 3 above)
Comment (arroyc) [reply build/__dotNetCoreRunTimeVersions.sh:7]: will do

File: src/BuildScriptGenerator/PythonVersions.cs [hunk 4]
```diff
@@ -9,6 +9,7 @@ public static class PythonVersions
         public const string Python27Version = "2.7.17";
         public const string Python36Version = "3.6.10";
         public const string Python37Version = "3.7.7";
-        public const string Python38Version = "3.8.2";
+        public const string Python38Version = "3.8.3";
+        public const string Python39Version = "3.9.0a6";
```
Comment (dorfire) [inline src/BuildScriptGenerator/PythonVersions.cs:13]: b1?

Comment (arroyc) [reply src/BuildScriptGenerator/PythonVersions.cs:13]: I would wait for b1 little bit ... a6 got released a while ago, b1 yesterday ... so it's more safe to have a6 release for now .. we will release b1 a bit later

File: images/build/Dockerfiles/Dockerfile [hunk 5]
```diff
@@ -184,15 +184,22 @@ ENV PYTHONIOENCODING UTF-8
...
     && ./installPlatform.sh python $PYTHON27_VERSION \
-    && ./installPlatform.sh python $PYTHON36_VERSION
+    && ./installPlatform.sh python $PYTHON36_VERSION \
+    && ./installPlatform.sh python $PYTHON38_VERSION \
+    && ./installPlatform.sh python $PYTHON39_VERSION
 RUN . ${BUILD_DIR}/__pythonVersions.sh && set -ex \
  && [ -d "/opt/python/$PYTHON27_VERSION" ] && echo /opt/python/$PYTHON27_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
  && [ -d "/opt/python/$PYTHON36_VERSION" ] && echo /opt/python/$PYTHON36_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
+ && [ -d "/opt/python/$PYTHON27_VERSION" ] && echo /opt/python/$PYTHON38_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
+ && [ -d "/opt/python/$PYTHON36_VERSION" ] && echo /opt/python/$PYTHON39_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
```
Comment (qianz2) [inline images/build/Dockerfiles/Dockerfile:194]: I think here versions need to be matched

Comment (arroyc) [reply images/build/Dockerfiles/Dockerfile:194]: good catch

File: src/BuildScriptGenerator/PythonVersions.cs (hunk 4 above)
Comment (arroyc) [reply src/BuildScriptGenerator/PythonVersions.cs]: updated

File: .gitignore [hunk 6]
```diff
@@ -256,10 +256,18 @@ paket-files/
 
 # Node and .NET Core runtime Dockerfiles, that are auto generated
 images/runtime/node/*/Dockerfile
+images/runtime/node/*/buster.Dockerfile
```
Comment (kichalla) [inline .gitignore:259]: revert these changes?

File: images/build/Dockerfiles/Dockerfile [hunk 7]
```diff
@@ -184,15 +184,22 @@ ENV PYTHONIOENCODING UTF-8
...
+ && [ -d "/opt/python/$PYTHON39_VERSION" ] && echo /opt/python/$PYTHON39_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
  && ldconfig
 RUN . ${BUILD_DIR}/__pythonVersions.sh && set -ex \
  && ln -s $PYTHON27_VERSION /opt/python/2.7 \
  && ln -s 2.7 /opt/python/2 \
- && ln -s $PYTHON36_VERSION /opt/python/3.6
+ && ln -s $PYTHON36_VERSION /opt/python/3.6 \
+ && ln -s $PYTHON36_VERSION /opt/python/3.7 \
+ && ln -s $PYTHON36_VERSION /opt/python/3.8 \
+ && ln -s $PYTHON36_VERSION /opt/python/3.9
```
Comment (qianz2) [inline images/build/Dockerfiles/Dockerfile:202]: Here as well.
